src/placeos-driver/driver_model.cr

```diff
@@ -40,6 +56,51 @@ struct PlaceOS::Driver::DriverModel
     property security : Hash(String, Array(String))
     # JSON Schema derived from the settings used in the driver
     property settings : NamedTuple(type: String, properties: Hash(String, JSON::Any)?, required: Array(String)?)?
+
+    # Note:: the use of both these functions is temporary
+    @[Deprecated("Use `#interface` instead")]
+    def functions : Hash(String, Hash(String, Array(JSON::Any)))
+      funcs = @functions
+      return funcs if funcs
+
+      funcs = {} of String => Hash(String, Array(JSON::Any))
+      @interface.not_nil!.each do |func_name, arguments|
+        arg_hash = {} of String => Array(JSON::Any)
+        arguments.each do |arg_name, schema|
+          values = [schema]
+          default = schema["default"]?
+          values << default if default
+
+          arg_hash[arg_name] = values
+        end
+        funcs[func_name] = arg_hash
+      end
+
+      @functions = funcs
+    end
+
+    def interface : Hash(String, Hash(String, JSON::Any))
+      iface = @interface
+      return iface if iface
+
+      Log.warn { "Deprecated metadata in use. A driver update is required" }
```

Comment:
Is this warning supposed to be here, or in `#functions`?

Reply:
this is meant to be here as it means the `@interface` object is nil and hence the class, which may have been initialized via JSON deserialising, is using the old version - there was no way to make this notify at compile time